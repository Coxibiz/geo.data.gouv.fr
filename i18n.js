import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'

import FooterTrad from './src/components/Footer/FooterTrad.json'
import LanguageSelectionTrad from './src/components/LanguageSelection/LanguageSelectionTrad.json'
import NewsletterFormTrad from './src/components/Newsletter/NewsletterFormTrad.json'

i18n
  .use(LanguageDetector)
  .init({
    // we init with resources
    resources: {
      en: {
        Footer: FooterTrad.en,
        LanguageSelection: LanguageSelectionTrad.en,
        NewsletterForm: NewsletterFormTrad.en
      },
      fr: {
        Footer: FooterTrad.fr,
        LanguageSelection: LanguageSelectionTrad.fr,
        NewsletterForm: NewsletterFormTrad.fr
      }
    },
    fallbackLng: 'en',

    // have a common namespace used around the full app
    ns: ['translations'],
    defaultNS: 'translations',

    keySeparator: false, // we use content as keys

    interpolation: {
      formatSeparator: ','
    },

    react: {
      wait: true
    }
  })

export default i18n
