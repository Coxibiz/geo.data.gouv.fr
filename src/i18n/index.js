import i18next from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import moment from 'moment'

const availableLanguages = [
  'en',
  'fr'
]

export default () => {
  const i18n = i18next
    .use(LanguageDetector)
    .init({
      resources: {
        en: {
          Common: require('../locales/en.json')
        },
        fr: {
          Common: require('../locales/fr.json')
        }
      },

      whitelist: [
        ...availableLanguages
      ],

      fallbackLng: 'en',

      defaultNS: 'Common',

      interpolation: {
        formatSeparator: ','
      },

      react: {
        wait: true
      }
    })

  i18n.availableLanguages = availableLanguages
  moment.locale(i18n.language)

  return i18n
}
