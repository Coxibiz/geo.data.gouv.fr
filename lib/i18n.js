import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import XHR from 'i18next-xhr-backend'
import moment from 'moment'

const dev = process.env.NODE_ENV !== 'production'

const languages = [
  'fr',
  'en'
]

if (!i18n.isInitialized) {
  i18n
    .use(XHR)
    .use(LanguageDetector)
    .init({
      fallbackLng: 'fr',
      load: 'languageOnly',
      lowerCaseLng: true,

      ns: ['common'],
      defaultNS: 'common',
      whitelist: [...languages],

      saveMissing: dev,

      interpolation: {
        escapeValue: false,
        formatSeparator: ',',
        format: (value, format, lng) => {
          if (format === 'uppercase') {
            return value.toUpperCase()
          }

          return value
        }
      },

      detection: {
        caches: ['cookie'],
        order: []
      }
    })

  i18n.on('languageChanged', lang => {
    moment.locale(lang)
  })
}

export default i18n
