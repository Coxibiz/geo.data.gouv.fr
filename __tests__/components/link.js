import React from 'react'
import renderer from 'react-test-renderer'
import i18n from 'i18next'
import {I18nextProvider} from 'react-i18next'

import Link from '../../components/link'

const i18next = i18n.init({ // eslint-disable-line import/no-named-as-default-member
  load: 'languageOnly'
})

describe('<Link />', () => {
  it('should render as french', () => {
    const inst = i18next.cloneInstance()
    inst.changeLanguage('fr')

    const tree = renderer
      .create(
        <I18nextProvider i18n={inst}>
          <Link href='/foo'>
            <a>Go somewhere</a>
          </Link>
        </I18nextProvider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render the as property', () => {
    const inst = i18next.cloneInstance()
    inst.changeLanguage('de')

    const tree = renderer
      .create(
        <I18nextProvider i18n={inst}>
          <Link href='/foo' as='/bar'>
            <a>Go somewhere</a>
          </Link>
        </I18nextProvider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should use the default language', () => {
    const inst = i18next.cloneInstance()

    const tree = renderer
      .create(
        <I18nextProvider initialLanguage='ru' i18n={inst}>
          <Link href='/foo'>
            <a>Go somewhere</a>
          </Link>
        </I18nextProvider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
