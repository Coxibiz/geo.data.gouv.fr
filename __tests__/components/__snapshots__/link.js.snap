// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> should render as french 1`] = `
<a
  href="/fr/foo"
  onClick={[Function]}
>
  Go somewhere
</a>
`;

exports[`<Link /> should render the as property 1`] = `
<a
  href="/de/bar"
  onClick={[Function]}
>
  Go somewhere
</a>
`;

exports[`<Link /> should use the default language 1`] = `
<a
  href="/ru/foo"
  onClick={[Function]}
>
  Go somewhere
</a>
`;
